src/main/java/io/harness/cf/client/api/CfClient.java

```diff
@@ -65,7 +65,10 @@ public CfClient(String apiKey, Config config) {
     // try to authenticate
     AuthService authService =
         new AuthService(defaultApi, apiKey, this, config.getPollIntervalInSec());
-    authService.startAsync();
+
+    // This runs oneIteration, instead of dealing with it as a async startup.
+    //
+    authService.runOneIteration();
```

Comment:
This calls runOneIteration which performs the authentication.  It can throw CfClientException.   This instead of using the startAsync call that runs in the background.   The problem with this approach is that Authentication can still fail, so we want to go for this approach we should check for errors as well as exceptions.